samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinator.java

```diff
@@ -296,15 +288,19 @@ void doOnProcessorChange() {
   /**
    * Stores the configuration of the job in the coordinator stream.
    */
-  private void storeConfigInCoordinatorStream() {
+  private void loadMetadataResources(JobModel jobModel) {
     MetadataStore metadataStore = null;
     try {
       // Creates the coordinator stream if it does not exists.
       createCoordinatorStream();
-
       MetadataStoreFactory metadataStoreFactory = Util.getObj(new JobConfig(config).getMetadataStoreFactory(), MetadataStoreFactory.class);
       metadataStore = metadataStoreFactory.getMetadataStore(SetConfig.TYPE, config, metrics.getMetricsRegistry());
       metadataStore.init();
+
+      MetadataResourceManager metadataResourceManager =
```

Comment:
Minor: Just to keep the same ordering, could this logic to above `createCoordinatorStream`? Maybe it would be slightly better since then the coordinator stream is created right before its usage (e.g. if `metadataResourceManager` fails, then we can skip trying to create the coord stream).

Reply:
Unfortunately no. The metadata store assumes the coordinator stream is created before being initialized. It does seem a little disconnected, but it's an issue with the metadata store design and implementation. I'll create a ticket to note this for the continued metadata store work.